app/main.py

```diff
@@ -36,17 +36,37 @@ async def upload_files(request: Request, session: SessionDep):
     files = body["files"]
     filesMimes = body["filesMimes"]
     filesNames = body["filesNames"]
-    file_result: dict[str, MarcFile] = {}
+    file_result: dict[str, IncompleteFile] = {}
 
     for file, mime, name in zip(files, filesMimes, filesNames):
-        decoded_bytes = b64decode(file)
-        entries = parse_marc(decoded_bytes)
-        file_result[name] = {
-            "mime": mime,
-            "size": naturalsize(len(decoded_bytes)),
-            "entries": entries,
-        }
-        incomplete.insert_incompletes(session, entries)
+        decoded_bytes: bytes = b64decode(file)
+        if mime == "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet":
+            entries = parse_excel(decoded_bytes)
+
+            file_result[name] = {
+                "mime": mime,
+                "size": naturalsize(len(decoded_bytes)),
+                "entries": entries,
+            }
+            incomplete.insert_incompletes(session, entries)
+        elif mime == "marc_file":
```

Comment:
nit: I like `if` here more than `elif`, since you there can be only one file type for a file. Also I don't think `marc_file` is correct value.

Reply:
I see your logic, but unless python works in mystical ways with else ifs the mime type will not be checked as many times unless it's onyx. If it was say excel it would check once and skip the other ones. With if statements, the mime type will be checked 3 times every single time.
You are right about the Marc mime type I left it there a as placeholder and forgot to replace it with the real value. Fixed.

Reply:
I would say unless you are working in a fairly low level environment like embedded system, if statement probably won't have a significant effect on performance. So this is more of a code style than anything. But it is nit so not a big deal

Reply:
yeah it's not gonna turn it from O(1) to O(n^3), it's like you said more semantic than anything.